packages/flutter/lib/src/material/list_item.dart

```diff
@@ -41,6 +41,35 @@ class ListItem extends StatelessComponent {
   final GestureTapCallback onTap;
   final GestureLongPressCallback onLongPress;
 
+  /// Add a one pixel border in between each item. If color isn't specified the
+  /// dividerColor of the context's theme is used.
+  static List<Widget> divideItems({ BuildContext context, Iterable<Widget> items, Color color }) {
```

Comment:
you can make it return an iterable really easily just by annotating the function as sync*, and then wherever you would normally call add(foo) just call yield foo.